### PR TITLE
docs: Add structure for developer documentation

### PR DESCRIPTION
**Pull request by thisthat**
Signed-off-by: Giovanni Liva <giovanni.liva@dynatrace.com>

This PR adds a structure to our developer documentation

### CODE REVIEW

File: docs/core.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Keptn Core Developer Docs
+
+
+This file contains helful resources to start developing with Keptn core services.
```
Comment (bacherfl) [inline docs/core.md:4]: helful -> helpful

File: docs/README.md [hunk 2]
```diff
@@ -20,24 +31,34 @@ While this is not a requirement, we recommend you to use any of the following
...
 
-If you want to contribute to the website or docs provided on the website, the 
+If you want to contribute to the website or docs provided on the website, the
  [keptn/keptn.github.io](https://github.com/keptn/keptn.github.io/) is the way to go.
 
-Last but not least, we have a collection of additional services at 
+Last but not least, we have a collection of additional services at
 * [github.com/keptn-contrib](https://github.com/keptn-contrib):
     * [dynatrace-service](https://github.com/keptn-contrib/dynatrace-service)
     * [dynatrace-sli-service](https://github.com/keptn-contrib/dynatrace-sli-service)
```
[the remark this answers is not in the document]
Comment (thisthat) [reply docs/README.md:64]: I would remove this list and maintain only the pointer to the two orgs. I think it is not maintainable as a list 😓

File: docs/README.md [hunk 3]
```diff
@@ -57,24 +78,24 @@ We aim to follow some rules for branch names within our repositories:
...
-For instance, if a new feature or bug branch is created, the workflow is to create a new branch of the **master** 
- branch, and  name it according to the convention listed above. Once ready, a new Pull Request should be created with 
+For instance, if a new feature or bug branch is created, the workflow is to create a new branch of the **master**
+ branch, and  name it according to the convention listed above. Once ready, a new Pull Request should be created with
  the **master** branch as a target. Hotfixes work similar, but should be branched away from the **master** branch. PRs
  for hotfixes should be created to the **master** and respective **release-** branches, ensuring that the latest release
  and the current development version use the fix.
-  
+
 ![Branch Workflow](assets/git_branch_workflow.png "Git Branch Workflow")
```
[the remark this answers is not in the document]
Comment (thisthat) [reply docs/README.md:91]: I agree. I'll add on the top a link towards that file

File: docs/README.md [hunk 4]
```diff
@@ -1,7 +1,23 @@
 # Keptn Developer Docs
 
-This folder contains docs for developers. If you are looking for the usage documentation of Keptn, or the `keptn` CLI, 
- please take a look at the [keptn.sh](https://keptn.sh/docs/) website.
+This folder contains docs for developers. If you are looking for the usage documentation of Keptn, or the `keptn` CLI,
+please take a look at the [keptn.sh](https://keptn.sh/docs/) website.
+We reccomend to read our [CONTRIBUTING](../CONTRIBUTING.md) guidelines before reading this document.
```
Comment (mowies) [inline docs/README.md:5]:
```suggestion
We recommend to read our [CONTRIBUTING](../CONTRIBUTING.md) guidelines before reading this document.
```

**Review by mowies — APPROVED** [review]

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
I would suggest adding explicit dev prefix or directory to the developer documentation. Later we could move user documentation to the repository too once we have Documentation-as-code in place.

I suggest moving all developer docs to `docs/developer` right away

**Comment by thisthat** [comment]
I agree with you @oleg-nenashev. I moved the doc into a `/docs/developer` subfolder :)

**Review by oleg-nenashev — APPROVED** [review]
Looks good, thank you for the patch!

FWIW GitHub Web UI now supports table of contents out of the box these days. The current approach is still better for visualization though

**Pull request merged**